surfactant/plugin/hookspecs.py

```diff
@@ -49,6 +50,7 @@ def extract_file_info(
         filetype (str): File type information based on magic bytes.
         context (Queue[ContextEntry]): Modifiable queue of entries from input config file. Existing plugins should still work without adding this parameter.
         children (List[Software]): List of additional software entries to include in the SBOM. Plugins can add additional entries, though if the plugin extracts files to a temporary directory, the context argument should be used to have Surfactant process the files instead.
+        include_all_files (bool): If all files should be included in the SBOM, rather than only those that are identified by Surfactant.
```

Comment:
```suggestion
        include_all_files (bool): If all files should be included in the SBOM, rather than only those with types that are recognized by Surfactant. When a plugin is adding additional context entries to the queue, it should typically default to propagating this value to the new context entries that it creates.
```

I hadn't considered that info extraction plugins would care about knowing this parameter... not entirely sure about the wording here, but (in addition to propagating the value to new context entries) there are also plugins that populate a list of additional children software entries to return, and their behavior change based on this argument... I'm just not sure how they are supposed to know which ones are of types that Surfactant would have identified.